src/modules/report/base.py

```diff
@@ -0,0 +1,34 @@
+from collector import services, vulnerabilities, services_lock, vulnerabilities_lock
+
+class BaseReporter(object):
+    def get_nodes(self):
+        nodes = list()
+        node_locations = set()
+        services_lock.acquire()
+        for service in services:
+            node_location = str(service.host)
+            if node_location not in node_locations:
+                nodes.append({"type": "Node/Master", "location": str(service.host)})
+                node_locations.add(node_location)
+        services_lock.release()
+        return nodes
+
+    def get_services(self):
+        services_lock.acquire()
+        services_data = [{"service": service.get_name(),
+                 "location": "{}:{}{}".format(service.host, service.port, service.get_path()),
+                 "description": service.explain()}
+                for service in services]
+        services_lock.release()
+        return services_data
+
+    def get_vulenrabilities(self):
```

Comment:
I know this typo already existed but while you're here, would you mind correcting the spelling in this function's name to `get_vulnerabilities()`?

Reply:
Done